src/Controller/v1/PaymentsController.ts

```diff
@@ -126,9 +126,9 @@ export class PaymentsController extends BaseHttpController {
   async refunds(request: Request, response: Response): Promise<void> {
     await this.httpService.callPaymentsServer(request, response, 'api/refunds', request.body)
   }
-  
-  @httpGet('/purchase', TYPES.AuthMiddleware)
+
+  @httpGet('/purchase', TYPES.EphemeralAuthMiddleware)
   async getPurchasePage(request: Request, response: Response): Promise<void> {
-    await this.httpService.callPaymentsServer(request, response, `api/purchase?sn_api_authorization=${request.query.sn_api_authorization}`)
+    await this.httpService.callPaymentsServer(request, response, `api/purchase?ephemeral_token=${request.query.ephemeral_token}`)
```

Comment:
Now that we have an endpoint to get the token maybe I could make it so that the website flow requests the token again before making a request, wdyt? Probably safer than having the token in the url

Reply:
Makes sense :)